### DISCUSSION TITLE
prisma generate ignored some fields.

### DISCUSSION THREAD

**Discussion by LukNasc** [prisma/prisma#4950] [Q&A] [answered]
I have the PullRequest and Repository model, when executing `npx prisma generate,`
the `repositoryId` field is ignored and the `Repository` field is needed to create new data. I just want to fill the `repositoryId` field with the repository id, not to create new data.

PullRequest model
```prisma
model PullRequest {
  id           Int         @id @default(autoincrement())
  title        String
  description  String?
  origin       String
  destination  String
  status       Int
  index        Int         @default(0)
  hash         String
  idRepository Int
  Repository   Repository  @relation(fields: [idRepository], references: [id])
  Reviewers    Reviewers[]

  @@index([idRepository], name: "idRepository")
}
```

Repository model
```prisma
model Repository {
  id          Int           @id @default(autoincrement())
  name        String
  description String?
  language    String?
  PullRequest PullRequest[]
}
```

Can someone help me please?

**Comment by ryands17** [comment, marked as answer]
Hey @LukNasc 👋 

If I understand correctly. you just want to pass the `repositoryId` while creating a `PullRequest`. In this case there are 2 ways to do this:

1. [Directly passing the `repositoryId`](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-new-profile-record-and-connect-it-to-an-existing-user-record):

For this, you would need to add the following to your `schema.prisma` and the run `prisma generate`.

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["uncheckedScalarInputs"]
}
```

Then you can easily create a `PullRequest` as follows:

```ts
await prisma.pullRequest.create({
    data: {
      destination: 'destination',
      hash: 'hash',
      origin: 'origin',
      status: 1,
      title: 'title',
      idRepository: 1,
    },
})
```

2. [Using the `connect` method that Prisma provides](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-new-profile-record-and-connect-it-to-an-existing-user-record):

```ts
await prisma.pullRequest.create({
    data: {
      destination: 'destination',
      hash: 'hash',
      origin: 'origin',
      status: 1,
      title: 'title',
      Repository: { connect: { id: 1 } },
    },
})
```

Both of the above approaches are equivalent and you can use any one of them. I have also posted the links to the docs so you can check that out as well. Let me know if you have any other questions.

**Reply by LukNasc** [reply]
Hi @ryands17 .
This is exactly what I wanted.
The second way was the one I used, I found the most correct one to do.
Thank you for your help. Here is the answer for those who have the same question.
Hug.

**Comment by LukNasc** [comment]
I did it exactly that way.

```prisma

await prisma.pullRequest.create({
    data: {
      destination: 'destination',
      hash: 'hash',
      origin: 'origin',
      status: 1,
      title: 'title',
      Repository: { connect: { id: 1 } },
    },
})
```

Thanks @ryands17

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
